### PR TITLE
add script to count shape inference implementations

### PR DESCRIPTION
**Pull request by anderspapitto**
at the moment, this outputs

```
103 operators have a type/shape inference function.
12 do not. These are:
 ATen
 ConvTranspose
 Crop
 FC
 GRU
 GRUUnit
 If
 LSTM
 Loop
 LoopIndexTensor
 RNN
 Upsample
```

### CODE REVIEW

**Comment by anderspapitto** [comment]
We could commit this and perhaps add the output into the docs somewhere, so that we stay up-to-date as more operators are added, or I can just keep it around as a one-off patch to run manually

File: onnx/defs/gen_shape_inference_information.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from __future__ import print_function
+
+from collections import defaultdict
```
Comment (bddppq) [inline onnx/defs/gen_shape_inference_information.py]: not used

File: onnx/defs/gen_shape_inference_information.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from __future__ import print_function
```
Comment (bddppq) [inline onnx/defs/gen_shape_inference_information.py]: In general all python files should have these boilerplate lines at the top

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals
```

File: onnx/defs/gen_shape_inference_information.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+from onnx import defs
+
+def main():
+    # domain -> support level -> name -> [schema]
+    with_inference = []
+    without_inference = []
+    for schema in defs.get_all_schemas():
+        domain, name, has_inference = schema.domain, schema.name, schema.has_type_and_shape_inference_function
+        (with_inference if has_inference else without_inference).append((domain, name))
```
Comment (bddppq) [inline onnx/defs/gen_shape_inference_information.py]: nit: too magic

File: onnx/defs/shape_inference.h [hunk 4]
```diff
@@ -17,6 +17,8 @@ struct InferenceContext {
 
 typedef void (*InferenceFunction)(InferenceContext&);
 
+inline void dummyInferenceFunction(InferenceContext&) { };
```
Comment (bddppq) [inline onnx/defs/shape_inference.h]: Maybe add a comment saying this is the default one that intend to do nothing.

**Pull request merged**